packages/mui-base/src/Checkbox/Root/CheckboxRoot.tsx

```diff
@@ -98,6 +98,7 @@ const CheckboxRoot = React.forwardRef(function CheckboxRoot(
     extraProps: {
       ...otherProps,
       ...otherGroupProps,
+      disabled,
```

Comment:
Instead of passing `disabled` here, it should be set by `useCheckboxRoot`'s `getButtonProps`. You can remove `aria-disabled` and have just `disabled` prop returned from there.
Same with the Switch.

Reply:
Thank you for pointing that out! 🙏 I’ve updated the code.
Please let me know if there are any further changes needed. 😊

Reply:
It appears that removing aria-disabled is causing CircleCI’s browser tests to fail. I suspect this might be an accessibility (a11y) issue. Do you think we should reintroduce aria-disabled alongside the disabled prop?

Reply:
Based on the various a11y resources, we are not required to use aria-disabled if we already are using native form elements like button with appropriate roles like switch or checkbox. what do you think @michaldudak ?

Reply:
No, having just the `disabled` attribute is enough. Using both `disabled` and `aria-disabled` is redundant and may trigger warnings in some a11y validators.
IMO the way to go would be updating the tests to expect the `disabled` attribute.

Reply:
Hi @michaldudak, I’m noticing a different assertion-related failure in CircleCI at the moment. Could you please check it and re-run the test for me, as I don’t have the access to do it myself?

Reply:
@michaldudak I have fixed the tests, Please review it again. Thanks !